site/_posts/2020-08-06-securely-implement-oauth-vuejs.md

```diff
@@ -1174,32 +1240,344 @@ methods: {
+
+In the `container` div we will update the `Greet` and `Login` components:
+
+```vue
+<Greet v-bind:email="email" v-bind:authState="authState" />
+<Login v-bind:email="email" v-bind:showSignout="showSignout" />
+```
+
+In the data function, we will add `showSignout` and `authState`:
+
+```vue
+data() {
+  return {
+    email: null,
+    body: null,
+    showSignout: false,
+    authState: null
+  }
+},
+```
+
+In the mounted function we are going to add `authState` and `showSignout`:
+
+```vue
+mounted() {
+  fetch(`http://localhost:9000/user`, {
+    credentials: "include" // fetch won't send cookies unless you set credentials
+  })
+  .then((response) => response.json())
+  .then((data) => {
+    if (data.authState == "Authorized"){
+      this.email = data.introspectResponse.email;
+      this.body = data.body;
+      this.showSignout = true
+    }
+    else if (data.authState == "notAuthorized"){
+      this.showSignout = true
+    }
+    else if (data.authState == "notAuthenticated"){
+      this.showSignout = false
+    }
+    this.authState = data.authState
+  });
+}
+```
+
+#### Add to your `Greeting.vue`
+
+Additionally, we will add `authState` to `Greeting.vue` (as well as some small styling):
+
+```vue
+<template>
+  <div className="greet">
+    <h3 v-if="email">Welcome{% raw %} {{email}} {% endraw %}</h3>
+    <h3 v-else>You are not logged in</h3>
+    <div>
+      <u>authState:</u>
+    </div>
+    <div class="authStateBox">{% raw %} {{authState}} {% endraw %}</div>
+  </div>
+</template>
+<script>
+export default {
+  name: 'Greet',
+  props: ["email", "authState"],
+};
+</script>
+<style>
+* {
+  margin-top: 30px;
+  text-align: center;
+  font-size: 20px;
+  font-family: 'Courier New', Courier, monospace;
+}
+
+.authStateBox {
+  margin: 20px;
+  background-color: lightcoral;
+  border-radius: 25px;
+}
+</style>
+```
+
+#### Add to your `Login.vue`
+In our `Login.vue` we will update to conditionally render on the `showSignout`:
+
+```vue
+<template>
+  <h1 v-if="showSignout"><a href='http://localhost:9000/logout'>Sign Out</a></h1>
+  <h1 v-else><a href='http://localhost:9000/login'>Sign In</a></h1>
+</template>
+<script>
+export default {
+  name: "Login",
+  props: ["showSignout"],
+};
+</script>
+```
+
+With this code (and a small amount of styling code for a logo) our application should look as follows:
+
+#### Authorized
+
+{% include _callout-note.liquid content=
+"The user is fully known to FusionAuth and the application."
+%}
+
+{% include _image.liquid src="/assets/img/blogs/oauth-vuejs/user-authorized.png" alt="User authorized and authenticated." class="img-fluid" figure=false %}
+
+{% include _image.liquid src="/assets/img/blogs/oauth-vuejs/user-configuration-after.png" alt="User settings for the new application." class="img-fluid" figure=false %}
+
+#### notAuthorized
+
+{% include _callout-note.liquid content=
```

Comment:
This feels like this is normal text, not a call out. Same with the other two sections here. Please change back to normal text.